ts/Grid/Pro/Export/Exporting.ts

```diff
@@ -0,0 +1,290 @@
+/* *
+ *
+ *  Grid Exporting class
+ *
+ *  (c) 2020-2025 Highsoft AS
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ *  Authors:
+ *  - Karol Kolodziej
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type Grid from '../../Core/Grid';
+import type { ExportingOptions } from '../../Core/Options';
+
+import U from '../../../Core/Utilities.js';
+import DownloadURL from '../../../Shared/DownloadURL.js';
+import Globals from '../../Core/Globals.js';
+
+const { downloadURL } = DownloadURL;
+const { merge } = U;
+const { win } = Globals;
+
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * Export the given table to CSV format.
+ */
+class Exporting {
+
+    /* *
+    *
+    *  Properties
+    *
+    * */
+
+    /**
+     * The Grid instance.
+     */
+    public readonly grid: Grid;
+
+    /**
+     * The options for the exporting.
+     */
+    public options: ExportingOptions;
+
+    /**
+     * Default options of the credits.
+     */
+    public static defaultOptions: ExportingOptions = {
+        firstRowAsNames: true,
+        itemDelimiter: ',',
+        lineDelimiter: '\n',
+        useLocalDecimalPoint: true
+    };
+
+
+    /* *
+    *
+    *  Constructor
+    *
+    * */
+
+    /**
+     * Construct the exporting.
+     *
+     * @param grid
+     * The Grid instance.
+     *
+     * @param options
+     * The options for the exporting.
+     */
+    constructor(grid: Grid, options?: ExportingOptions) {
+        this.grid = grid;
+        this.options = merge(Exporting.defaultOptions, options);
+    }
+
+
+    /* *
+    *
+    *  Methods
+    *
+    * */
+
+    /**
+     * Downloads the CSV string as a file.
+     */
+    public downloadCSV(): void {
+        const csv = this.getCSV();
+
+        downloadURL(
+            this.getBlobFromContent(csv, 'text/csv') ||
+                'data:text/csv,\uFEFF' + encodeURIComponent(csv),
+            this.getFilename() + '.csv'
+        );
+    }
```

Comment:
Do we need to duplicate the method, because in https://code.highcharts.com/modules/export-data.src.js there is exactly the same method.